cmd/meroxa/flink/deploy.go

```diff
@@ -0,0 +1,96 @@
+package flink
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/meroxa/cli/log"
+	"github.com/meroxa/turbine-core/pkg/ir"
+
+	"golang.org/x/mod/semver"
+)
+
+const (
+	irFilename       = "meroxa-ir.json"
+	exactJavaVersion = "v11"
+	modeEnvVar       = "MEROXA_PLATFORM"
+	outputEnvVar     = "MEROXA_OUTPUT"
+	irVal            = "EMIT_IR"
+)
+
+func GetIRSpec(ctx context.Context, jarPath string, secrets map[string]string, l log.Logger) (*ir.DeploymentSpec, error) {
+	verifyJavaVersion(ctx, l)
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	irFilepath := filepath.Join(cwd, irFilename)
+
+	cmd := exec.CommandContext(ctx, "java", "-jar", jarPath)
+	cmd.Env = append(
+		cmd.Environ(),
+		fmt.Sprintf("%s=%s", modeEnvVar, irVal),
+		fmt.Sprintf("%s=%s", outputEnvVar, irFilename))
+	_, err = cmd.CombinedOutput() // all java output goes to stderr, so that's fun
+	defer func() {
+		_ = os.Remove(irFilepath)
+	}()
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = os.Stat(irFilepath)
+	if err != nil {
+		// @TODO try the docker way because the jar is skinny
+		// Otherwise, there are no Meroxa* classes in this main class
+		return nil, nil
+	}
+
+	b, err := os.ReadFile(irFilepath)
+	if err != nil {
+		return nil, err
+	}
+
+	// @TODO assess the scope of updating validateCollections to use the ConnectorSpec
+	var spec ir.DeploymentSpec
+	err = json.Unmarshal(b, &spec)
+	if err != nil {
+		return nil, err
+	}
+
+	spec.Secrets = secrets
+	return &spec, nil
+}
+
+func verifyJavaVersion(ctx context.Context, l log.Logger) {
+	cmd := exec.CommandContext(ctx, "java", "-version")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		l.Warnf(ctx,
+			"warning: unable to verify local Java version is compatible with the Meroxa Platform; jar's must be compiled for %s",
+			exactJavaVersion)
+		return
+	}
+
+	// looks like 'openjdk version "11.0.19" 2023-04-18'
+	r := regexp.MustCompile(`version "([0-9.]+.[0-9.]+.[0-9.]+)"`)
```

Comment:
do we need to validate this _exact_ version? locally I have `openjdk64-11.0.18` installed, as well as the microsoft jdk 11

I wonder if we can generalize to just support java 11, though that is a bit more work

Reply:
Ah, I see that you're verifying `v11` in general. Though, it may not show in this exact format for the regex

Reply:
ok, I take this back, my local versions are matching this regex:
```
❯ java --version
openjdk 11.0.18 2023-01-17 LTS
OpenJDK Runtime Environment Microsoft-7208460 (build 11.0.18+10-LTS)
OpenJDK 64-Bit Server VM Microsoft-7208460 (build 11.0.18+10-LTS, mixed mode)
```

Reply:
it's a bad constant name... I'll update to majorJavaVersion.

Reply:
also, the output of -version and --version are slightly different :lolsob: